Sources/PathKit.swift

```diff
@@ -152,8 +152,18 @@ extension Path {
   ///
   public func abbreviate() -> Path {
 #if os(Linux)
-    // TODO: actually de-normalize the path
-    return self
+    let env = ProcessInfo.processInfo.environment
+    guard let home = env["HOME"], !home.isEmpty else { return self }
+    guard let homeRange = self.path.range(of: home, options: [.anchored, .caseInsensitive]) else { return self }
+    let withoutHome = Path(self.path.replacingCharacters(in: homeRange, with: ""))
+    
+    if withoutHome.path.isEmpty || withoutHome.path == "/" {
```

Comment:
Hm, good question. A couple thoughts:

1. Is freeBSD not caught by `if os(Linux)`? If so, then adding freeBSD would likely involve changes other than this line (e.g. the first line in the file).
1. We may end up in the same trap if apple decides to release a different OS.
1. I'm not really familiar enough to know if this would work on Windows or not? I believe cygwin defines a similar Linux `Home` structure but I don't know about Windows.

Given the above, I feel like, unless we have a suite of tests able to run on the supported platforms, we aren't able to make informed decisions about platform support. When we get those things, and as Apple adds more platforms, then I think we can revisit, and indeed the rest of the code requires a revisiting anyway. Finally, that doing it this way doesn't necessarily prevent us from supporting newer platforms in the future, and instead allows us to constrain which platforms have special behavior, with the default assumption that as new platforms are added, they will either be added with full Foundation support, or we'll need to handle them as special cases anyway.